MAINTAINERS.md

```diff
@@ -77,8 +87,66 @@ Make frequent project releases to the community.
+
+### Interest
+
+Upon receiving at least 3 positive maintainer votes, and no vetoes, from other maintainers within a maximum period of 2 weeks, the nominating maintainer asks a potential nominee whether they might be interested in becoming a maintainer on the repository, which typically happens via e-mail obtained via DCO, but may also be a comment to a pull request, following a substantial contribution.
+
+> This is great work! Based on your ongoing contributions to the project we have decided that we'd like to invite you to become a co-maintainer for this project. Please let us know whether you accept this nomination.
+
+Individuals accept the nomination by replying, or commenting, for example _"Thank you! I would love to."_
+
+### Public Confirmation
+
+Upon receiving 3 positive maintainer votes, and no vetoes, from other maintainers, and after having confirmed interest with the nominee, the maintainer opens a pull request adding the proposed co-maintainer to MAINTAINERS.md, and perform a final vote similar to the nomination process. If the nomination still has 3 positive votes, and no vetoes, from other maintainers, the nomination is accepted. The positive maintainer voters should approve the pull request to show support.
+
+> _Content from the above nomination._
+> 
+> Here is my +1.
+> 
+> If you support this nomination, please add your +1 or a comment. 
+> 
+> If you do not support this nomination, we invite you to e-mail all/some of the current [project maintainers](https://github.com/opensearch-project/OpenSearch/blob/main/MAINTAINERS.md) in the next 2 weeks. We will keep your feedback private to this group and discuss offline. We will not publicly post the reasons for a decision not to add this co-maintainer, or publicize your name. 
+> 
+> Anyone, including maintainers, can publicly add +1s as requested in the nomination issue if they support the nomination or contact the maintainers in private if they oppose. The pull request stays open for 2 weeks.
+
+### Maintainer Decision
+
+If a positive decision has been reached, the nomination is accepted. If the maintainers cannot reach a positive decision for any reason, the nomination is rejected with _"a decision could not be reached"_ with no further explanation necessary, and the pull request is closed.
+
+### Adding a New Maintainer
+
+The repo admin adjusts the new maintainer’s permissions accordingly, and merges the pull request to MAINTAINERS.md.
+
+## Removing a Maintainer
+
+Removing a maintainer is a disruptive action that the community of maintainers should not undertake lightly. There are several reasons a maintainer might be removed from the project, such as violating the [code of conduct](https://github.com/opensearch-project/.github/blob/main/CODE_OF_CONDUCT.md), or taking other actions that negatively impact the project.
```

Comment:
We should be explicit here about actions that we already know will result in loss of maintainer status. Violating code of conduct is one, but the "might be" language can lead to trouble. We may not know everything, but we know some things

Reply:
Do we need an explicit process for contributors and admins to start a removal?

Reply:
Agreed. I changed _might_ to _will_ in https://github.com/opensearch-project/.github/pull/59/commits/6d68d527c1e48033708f824843ecad4e27d51291.